core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -325,6 +325,10 @@ private[spark] class MemoryStore(
 
     // Whether there is still enough memory for us to continue unrolling this block
     var keepUnrolling = true
+    // Number of elements unrolled so far
+    var elementsUnrolled = 0L
+    // How often to check whether we need to request more memory
+    val memoryCheckPeriod = 16
```

Comment:
This is the same approach we used in `putIteratorAsValues`. One thing still missing is `memoryGrowthFactor`, can we add that too?

Reply:
in the future we should do some refactor between `putIteratorAsValues` and `putIteratorAsBytes`, to reduce duplicated code.

Reply:
OK, I'll add it. I think I can take a try to refactor it. Do you need a separate pr?

Reply:
yea definitely not in this PR.